uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -473,4 +524,51 @@ private function getContractDatesForContactInPeriod(AbsencePeriod $absencePeriod
 
     return $contractsDates;
   }
+
+	/**
+	 * Returns formatted result for getting the balance for an entitlement period given an
+	 * Entitlement Id or (Contact ID + Absence Period ID), When params contains the include_future parameter and its true,
+	 * It returns also future balance for an entitlement taking the Awaiting Approval and More Information Requested leave statuses
+	 * into consideration
+	 *
+	 * Sample param: $params = ['entitlement_id'=>1, 'contact_id'=>9, 'include_future'=>false]
+	 *
+	 * @param $params
+	 * @return an array of formatted results
+	 *
+	 * [
+	 * ['id'=>1, 'remainder'=>['current'=>30, 'future'=>20]]
+	 * ]
+	 */
+	public static function getLeavePeriodEntitlementRemainder($params){
+
```

Comment:
This readability of this method is very low with so many ifs, elses and indentation levels. In 99.99% of times you don't really need elses. Also, it's always nice to try to have only one or two levels of indentation. Here's an example of how this method can be rewritten applying these rules:

```php
$leavePeriodEntitlements = [];
if(!empty($params['entitlement_id'])) {
  $leavePeriodEntitlements[] = self::findById($params['entitlement_id']);
}

if(!empty($params['contact_id']) && !empty($params['period_id'])) {
  $leavePeriodEntitlements = self::getPeriodEntitlementsForContact($params['contact_id'], $params['period_id']);
}

$returnValues = [];
foreach($leavePeriodEntitlements as $leavePeriodEntitlement) {
  $remainder = ['current' => $leavePeriodEntitlement->getBalance()];
  
  if(!empty($params['include_future'])) {
  	$remainder['future'] => $leavePeriodEntitlement->getFutureBalance();
  }

  $returnValues[] = [
    'id' => $leavePeriodEntitlements->id,
    'remainder' => $remainder
  ];
}

return $returnValues;
```

Please note that I didn't test this and I don't even know if it works. I'm just putting it here as an example of how it can be improved